src/test/java/com/cdancy/bitbucket/rest/BaseBitbucketApiLiveTest.java

```diff
@@ -67,12 +70,28 @@ protected Properties setupProperties() {
         return overrides;
     }
 
-    protected synchronized String getDefaultUser() {
-        if (defaultUser == null) {
+    protected synchronized String getDefaultUserAsString() {
+        if (defaultUserAsString == null) {
             if (this.credential.contains(":")) {
-                defaultUser = this.credential.split(":")[0];
+                defaultUserAsString = this.credential.split(":")[0];
             } else {
-                defaultUser = new String(base64().decode(this.credential)).split(":")[0];
+                defaultUserAsString = new String(base64().decode(this.credential)).split(":")[0];
+            }
+        }
+        return defaultUserAsString;
+    }
+
+    protected synchronized User getDefaultUser() {
+        if (defaultUser == null) {
+            String username = getDefaultUserAsString();
+            final UserPage userPage = api.adminApi().listUserByGroup(defaultBitbucketGroup, null, null, null);
+            assertThat(userPage).isNotNull();
+            assertThat(userPage.size() > 0).isTrue();
+            for (User user : userPage.values()) {
+                if (username.equals(user.slug())) {
+                    defaultUser = user;
+                    break;
+                }
             }
```

Comment:
I would just return once we've found the correct user and throw an exception if somehow we get past the for loop.